uqcsbot/models.py

```diff
@@ -1,56 +1,56 @@
 from sqlalchemy import (
     BigInteger,
     Boolean,
-    Column,
     Date,
     DateTime,
     Integer,
     String,
     Time,
 )
+from typing import Optional
+from datetime import datetime
 
-Base = declarative_base()
 
-
-# Used for linking a message to a bot function.
-# Previously used for the channel cog, currently unused.
-class Message(Base):
-    __tablename__ = "messages"
-
-    id = Column("id", BigInteger, primary_key=True, nullable=False)
-    type = Column("type", String, nullable=False)
+class Base(DeclarativeBase):
+    pass
 
 
 class AOCWinner(Base):
     __tablename__ = "aoc_winner"
 
-    id = Column("id", BigInteger, primary_key=True, nullable=False)
-    aoc_userid = Column("aoc_userid", Integer, nullable=False)
-    year = Column("year", Integer, nullable=False)
+    id: Mapped[int] = mapped_column("id", BigInteger, primary_key=True, nullable=False)
+    aoc_userid: Mapped[int] = mapped_column("aoc_userid", Integer, nullable=False)
+    year: Mapped[int] = mapped_column("year", Integer, nullable=False)
 
 
 class MCWhitelist(Base):
     __tablename__ = "mc_whitelisted"
 
-    mc_username = Column("mcuser", String, primary_key=True, nullable=False)
-    discord_id = Column("discordid", BigInteger, nullable=False)
-    admin_whitelisted = Column("adminwl", Boolean)
-    added_dt = Column("added_dt", DateTime, nullable=False)
+    mc_username: Mapped[str] = mapped_column(
+        "mcuser", String, primary_key=True, nullable=False
+    )
+    discord_id: Mapped[str] = mapped_column("discordid", BigInteger, nullable=False)
+    admin_whitelisted: Mapped[Optional[bool]] = mapped_column("adminwl", Boolean)
+    added_dt: Mapped[datetime] = mapped_column("added_dt", DateTime, nullable=False)
 
 
 class Reminders(Base):
     __tablename__ = "reminders"
 
-    id = Column("id", BigInteger, primary_key=True, nullable=False)
-    user_id = Column("user_id", BigInteger, nullable=False)
-    channel_id = Column("channel_id", BigInteger, nullable=True)
-    time_created = Column("time_created", DateTime, nullable=False)
-    message = Column("message", String, nullable=False)
-    time = Column("time", Time, nullable=False)
-    start_date = Column("start_date", Date, nullable=False)
-    end_date = Column("end_date", Date, nullable=True)
-    week_frequency = Column("week_frequency", Integer, nullable=True)
+    id: Mapped[int] = mapped_column("id", BigInteger, primary_key=True, nullable=False)
+    user_id: Mapped[int] = mapped_column("user_id", BigInteger, nullable=False)
+    channel_id: Mapped[Optional[int]] = mapped_column(
+        "channel_id", BigInteger, nullable=True
+    )
+    time_created: Mapped[int] = mapped_column("time_created", DateTime, nullable=False)
+    message: Mapped[str] = mapped_column("message", String, nullable=False)
+    time = mapped_column("time", Time, nullable=False)
+    start_date = mapped_column("start_date", Date, nullable=False)
+    end_date = mapped_column("end_date", Date, nullable=True)
```

Comment:
I'm unsure of protocol: do these need types?

Reply:
I think they will in order to be used properly in other files, but the other files are excluded for now until I get the full PR done.